kornia/color/hsv.py

```diff
@@ -65,14 +65,14 @@ def hsv_to_rgb(image: torch.Tensor) -> torch.Tensor:
     q: torch.Tensor = v * (one - f * s)
     t: torch.Tensor = v * (one - (one - f) * s)
 
-    out: torch.Tensor = torch.stack([hi, hi, hi], dim=-3)
-
-    out[out == 0] = torch.stack((v, t, p), dim=-3)[out == 0]
-    out[out == 1] = torch.stack((q, v, p), dim=-3)[out == 1]
-    out[out == 2] = torch.stack((p, v, t), dim=-3)[out == 2]
-    out[out == 3] = torch.stack((p, q, v), dim=-3)[out == 3]
-    out[out == 4] = torch.stack((t, p, v), dim=-3)[out == 4]
-    out[out == 5] = torch.stack((v, p, q), dim=-3)[out == 5]
+    hi = hi.long()
+    indices: torch.Tensor = torch.stack([hi, hi + 6, hi + 12], dim=-3)
+    out = torch.stack((
+        v, q, p, p, t, v,
+        t, v, v, q, p, p,
+        p, p, t, v, v, q,
+    ), dim=-3)
+    out = torch.gather(out, -3, indices)
```

Comment:
is there any way to index directly from a (v,q,p,t) tensor instead of stacking repeated patterns ?